### PR TITLE
Azurestack admin module 1.2.12 migration guide

### PR DESCRIPTION
**Pull request by bganapa**
## Description

Stack admin module migration guide.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: documentation/migration-guides/Stack/migration-guide.1.2.12.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
+# Table of Contents
+1. [Summary](#summary)
+2. [Breaking changes in subscriptions module](#Breaking-changes-in-subscriptions-module)
+3. [Breaking changes in storage admin module](#Breaking-changes-in-storage-admin-module)
+
+## Summary
+The module version 1.2.12 of AzureStack brings in many breaking changes. Most of these breaking changes were noted with warnings in the previous releases. Please note that there are no changes in the AzureRM version of the module specific to azure stack.
```
Comment (deathly809) [inline documentation/migration-guides/Stack/migration-guide.1.2.12.md]: azure stack -> Azure Stack

File: documentation/migration-guides/Stack/migration-guide.1.2.12.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+**Parameter Changes**<br>
+The previous release was printing  warnings about the deprecation of the following parameter aliases and they are deprecated now. 
+- Parameter alias ```PlanName``` has been deprecated in favor of ```Name``` in Plan cmdlets
+- Parameter alias ```OfferName``` has been deprecated in favor of ```Name``` in Offer cmdlets
+- Parameter alias ```ResourceGroup``` has been deprecated in favor of ```ResourceGroupName``` in all the cmdlets
+- Parameter ```Managed``` in Get-AzsPlan has been deprecated. It was not used before
+- Parameter ```Managed``` in Get-AzsOffer has been deprecated. Please use Get-AzsManagedOffer instead
+
+**Flattening of Properties** <br>
+The Plan/Offer/Subscription objects wont be having .Properties instead the child properties are flattened out to be at the Top level. If you are having any references to the .Properties, this breaking change could be fixed by accessing the prperty directly at the top level
```
Comment (deathly809) [inline documentation/migration-guides/Stack/migration-guide.1.2.12.md]: prperty -> property

Comment (deathly809) [reply documentation/migration-guides/Stack/migration-guide.1.2.12.md]: Top -> top

Comment (deathly809) [reply documentation/migration-guides/Stack/migration-guide.1.2.12.md]: wont -> won't

File: documentation/migration-guides/Stack/migration-guide.1.2.12.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+**Globally Removed Parameters**<br>
+The parameters ```SkipCertificateValidation``` and ```DefaultProfile``` have been removed from all cmdlets.
+
+The parameters ```TimeGrain```, ```StartTimeInUtc```, ```EndTimInUtc```, ```MetricNames```, ```DetailedOutput```  have been removed for Get-Azs*Metric Cmdlets, please replace with Where-Object and Select-Object.
+
+The pareamters ```MetricNames```, ```DetailedOutput```  have been removed for Get-Azs*MetricDefinition Cmdlets, please replace with Where-Object and Select-Object
+
+**New Parameters**
+- The parameter ```ResourceGroupName``` is now an optional for all cmdlets.
```
Comment (deathly809) [inline documentation/migration-guides/Stack/migration-guide.1.2.12.md]: Except in subscriptions and AzureBridge!

File: documentation/migration-guides/Stack/migration-guide.1.2.12.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+- Get-AzsStorageShare
+- Get-AzsStorageShareMetric
+- Get-AzsStorageShareMetricDefinition
+- Get-AzsTableService
+- Get-AzsTableServiceMetric
+- Get-AzsTableServiceMetricDefinition
+- Start-AzsReclaimStorageCapacity
+
+
+-  ```ResourceId``` parameter can be used for cmdlets that require ```Name```
```
Comment (deathly809) [inline documentation/migration-guides/Stack/migration-guide.1.2.12.md]: Can you elaborate on this?  When I first read this I didn't understand right away.  Also, you should mention this has an Alias for 'id'.

File: documentation/migration-guides/Stack/migration-guide.1.2.12.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
+# Table of Contents
```
Comment (deathly809) [inline documentation/migration-guides/Stack/migration-guide.1.2.12.md]: Do we need a list of known issues?

File: documentation/migration-guides/Stack/migration-guide.1.2.12.md (hunk 1 above)
Comment (bganapa) [reply documentation/migration-guides/Stack/migration-guide.1.2.12.md]: fixed

File: documentation/migration-guides/Stack/migration-guide.1.2.12.md (hunk 5 above)
Comment (bganapa) [reply documentation/migration-guides/Stack/migration-guide.1.2.12.md]: known issues will be in the main documentation site. I think it makes sense to add it here as well. We can do that later. I still expect us to edit this for few details even after the release(like adding specific examples Old vs New )

File: documentation/migration-guides/Stack/migration-guide.1.2.12.md (hunk 4 above)
Comment (bganapa) [reply documentation/migration-guides/Stack/migration-guide.1.2.12.md]: added a line. agreed this is confusing, will be better if we add an example. I think we will have to do that later. Unlike subscription module, storage module was not used extensively by the customer

File: documentation/migration-guides/Stack/migration-guide.1.2.12.md (hunk 3 above)
Comment (deathly809) [reply documentation/migration-guides/Stack/migration-guide.1.2.12.md]: Okay!  Do we mention somewhere that all resource group names are optional except for those two?  Also, how do users find those resource groups?  I guess that is not for use to describe?

Comment (bganapa) [reply documentation/migration-guides/Stack/migration-guide.1.2.12.md]: don't think it needs to go in a separate doc, each cmdlet's reference doc should cover that

File: src/StackAdmin/Azs.AzureBridge.Admin/Help/Azs.AzureBridge.Admin.md [hunk 6]
```diff
@@ -1,16 +1,16 @@
 ---
-Module Name: Azs.AzureBridge.Admin
+Module Name: Azs.Azurebridge.Admin
 Module Guid: 82d2260a-95ae-44bb-af8b-afd67d38f6db
 Download Help Link: {{Please enter FwLink manually}}
```
Comment (deathly809) [inline src/StackAdmin/Azs.AzureBridge.Admin/Help/Azs.AzureBridge.Admin.md]: I don't know what this is.  Maybe Thomas does?

File: documentation/migration-guides/Stack/migration-guide.1.2.12.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
```
Comment (markcowl) [inline documentation/migration-guides/Stack/migration-guide.1.2.12.md]: We will wan to update the version numbers and name for this file

**Comment by bganapa** [comment]
@markcowl pushed a commit to rename the file and the contents to 1.3.0

**Pull request merged**